### PR TITLE
Update image digest and runtime GHAs to include the new ROCm changes

### PR DESCRIPTION
**Pull request by atheo89**
Related to: https://issues.redhat.com/browse/RHOAIENG-6445 

## Description
This PR refresh the downstream github actions to include the ROCm.

More specifically:
- On digest updater: 
  - Include the new ROCm minimal, pytorch, tensorflow images
  -  Revamp the code to be the as much close with upstream codebase

- On runtime updater:
  - update the action to work also for the new runtimes.
  
NOTE: I had to rearrange a bit the order on the params.env and commit.env files. 
  
## How Has Been Tested
Here it is a local repo that simulates the behaviour of the workflow: [link](https://github.com/atheo89/autorefresh-image-digest-workflow/actions/runs/10538598251/job/29201288123)

### CODE REVIEW

File: .github/workflows/notebook-digest-updater.yaml [hunk 1]
```diff
@@ -147,61 +162,68 @@ jobs:
...
             regex=${REGEXES[$i]}
-            img=$(cat manifests/base/params.env | grep -E "${image}=" | cut -d '=' -f2)
-            registry=$(echo $img | cut -d '@' -f1)
+            skopeo_metadata=$(skopeo inspect --retry-times 3 "docker://${img}")
             latest_tag=$(skopeo inspect docker://$img | jq -r --arg regex "$regex" '.RepoTags | map(select(. | test($regex))) | .[0]')
             digest=$(skopeo inspect docker://$registry:$latest_tag | jq .Digest | tr -d '"')
             output=$registry@$digest
             echo "NEW: " $output
             sed -i "s|${image}=.*|${image}=$output|" manifests/base/params.env
+            i=$((i+1))
```
Comment (jstourac) [inline .github/workflows/notebook-digest-updater.yaml:196]: Hm, why?

File: .github/workflows/notebook-digest-updater.yaml [hunk 2]
```diff
@@ -147,61 +162,68 @@ jobs:
...
+          i=0
+          for image in ${IMAGES}; do
+            echo "CHECKING: '${image}'"
+            img=$(grep -E "${image}=" "${PARAMS_ENV_PATH}" | cut -d '=' -f2)
+            registry=$(echo "${img}" | cut -d '@' -f1)
             regex=${REGEXES[$i]}
-            img=$(cat manifests/base/params.env | grep -E "${image}=" | cut -d '=' -f2)
-            registry=$(echo $img | cut -d '@' -f1)
+            skopeo_metadata=$(skopeo inspect --retry-times 3 "docker://${img}")
             latest_tag=$(skopeo inspect docker://$img | jq -r --arg regex "$regex" '.RepoTags | map(select(. | test($regex))) | .[0]')
```
Comment (jstourac) [inline .github/workflows/notebook-digest-updater.yaml:191]:
```suggestion
            latest_tag=$(echo "${skopeo_metadata}" | jq -r --arg regex "$regex" '.RepoTags | map(select(. | test($regex))) | .[0]')
```

File: .github/workflows/notebook-digest-updater.yaml [hunk 3]
```diff
@@ -65,61 +65,76 @@ jobs:
...
+          # The order of the regexes array should match with the params.env file
+          REGEXES=("v2-${{ env.RELEASE_VERSION_N }}-\d{8}-+${{ steps.hash-n.outputs.HASH_N }}" \
+                  "cuda-[a-z]+-minimal-[a-z0-9]+-[a-z]+-3.9-${{ env.RELEASE_VERSION_N }}-\d{8}-${{ steps.hash-n.outputs.HASH_N }}" \
+                  "v2-${{ env.RELEASE_VERSION_N }}-\d{8}-+${{ steps.hash-n.outputs.HASH_N }}" \
+                  "v2-${{ env.RELEASE_VERSION_N }}-\d{8}-+${{ steps.hash-n.outputs.HASH_N }}" \
+                  "cuda-[a-z]+-tensorflow-[a-z0-9]+-[a-z]+-3.9-${{ env.RELEASE_VERSION_N }}-\d{8}-${{ steps.hash-n.outputs.HASH_N }}" \
+                  "v2-${{ env.RELEASE_VERSION_N }}-\d{8}-+${{ steps.hash-n.outputs.HASH_N }}" \
+                  "codeserver-[a-z0-9]+-[a-z]+-3.9-${{ env.RELEASE_VERSION_N }}-\d{8}-${{ steps.hash-n.outputs.HASH_N }}" \
+                  "habana-[a-z]+-1.13.0-[a-z0-9]+-[a-z]+-3.8-${{ env.RELEASE_VERSION_N }}-\d{8}-${{ steps.hash-n.outputs.HASH_N }}" \
+                  "habana-[a-z]+-1.13.0-[a-z0-9]+-[a-z]+-3.8-${{ env.RELEASE_VERSION_N }}-\d{8}-${{ steps.hash-n.outputs.HASH_N }}" \
```
Comment (jstourac) [inline .github/workflows/notebook-digest-updater.yaml:86]: I think this line is duplicate and should be removed.

File: .github/workflows/notebook-digest-updater.yaml [hunk 4]
```diff
@@ -147,61 +162,68 @@ jobs:
...
+            registry=$(echo "${img}" | cut -d '@' -f1)
             regex=${REGEXES[$i]}
-            img=$(cat manifests/base/params.env | grep -E "${image}=" | cut -d '=' -f2)
-            registry=$(echo $img | cut -d '@' -f1)
+            skopeo_metadata=$(skopeo inspect --retry-times 3 "docker://${img}")
             latest_tag=$(skopeo inspect docker://$img | jq -r --arg regex "$regex" '.RepoTags | map(select(. | test($regex))) | .[0]')
             digest=$(skopeo inspect docker://$registry:$latest_tag | jq .Digest | tr -d '"')
             output=$registry@$digest
             echo "NEW: " $output
             sed -i "s|${image}=.*|${image}=$output|" manifests/base/params.env
```
Comment (jstourac) [inline .github/workflows/notebook-digest-updater.yaml:195]:
```suggestion
            digest=$(skopeo inspect --retry-times 3 "docker://${registry}:${latest_tag}" | jq .Digest | tr -d '"')
            output="${registry}@${digest}"
            echo "NEW: ${output}"
            sed -i "s|${image}=.*|${image}=${output}|" "${PARAMS_ENV_PATH}"
```

File: .github/workflows/notebook-digest-updater.yaml [hunk 5]
```diff
@@ -65,61 +65,76 @@ jobs:
...
-            img=$(cat manifests/base/params.env | grep -E "${image}=" | cut -d '=' -f2)
-            registry=$(echo $img | cut -d '@' -f1)
-            latest_tag=$(skopeo inspect docker://$img | jq -r --arg regex "$regex" '.RepoTags | map(select(. | test($regex))) | .[0]')
-            digest=$(skopeo inspect docker://$registry:$latest_tag | jq .Digest | tr -d '"')
-            output=$registry@$digest
-            echo "NEW: " $output
-            sed -i "s|${image}=.*|${image}=$output|" manifests/base/params.env
+            skopeo_metadata=$(skopeo inspect --retry-times 3 "docker://${img}")
+            latest_tag=$(skopeo inspect "docker://${img}" | jq -r --arg regex "$regex" '.RepoTags | map(select(. | test($regex))) | .[0]')
+            digest=$(skopeo inspect "docker://${registry}:${latest_tag}" | jq .Digest | tr -d '"')
```
Comment (jstourac) [inline .github/workflows/notebook-digest-updater.yaml:100]:
```suggestion
            digest=$(skopeo inspect --retry-times 3 "docker://${registry}:${latest_tag}" | jq .Digest | tr -d '"')
```

**Review by jstourac — COMMENTED** [review]
I put couple of comments, otherwise LGTM.

**Comment by atheo89** [comment]
Great catch Jan, thanks for the review :slightly_smiling_face: I push the suggestions.

**Comment by jstourac** [comment]
Thank you, LGTM now :slightly_smiling_face: 

/lgtm

**Pull request merged**